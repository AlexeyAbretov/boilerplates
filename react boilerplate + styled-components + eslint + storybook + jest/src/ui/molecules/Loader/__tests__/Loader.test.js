import React from 'react';
import renderer from 'react-test-renderer';

import { Loader, AbsoluteLoader } from '..';

describe('ui/atoms/Loader', () => {
  it('should be rendered', () => {
    const tree = renderer.create(<Loader isVisible />);
    expect(tree).toMatchSnapshot();
  });

  it('should be rendered with overlay', () => {
    const tree = renderer.create(<Loader isVisible  overlayProps={{}} />);
    expect(tree).toMatchSnapshot();
  });

  it('should be rendered as absolute', () => {
    const tree = renderer.create(<AbsoluteLoader isVisible  />);
    expect(tree).toMatchSnapshot();
  });

  test('should not be rendered without params', () => {
    const tree = renderer.create(<Loader />);
    expect(tree).toMatchSnapshot();
  });

  test('should not be rendered', () => {
    const tree = renderer.create(<Loader isVisible={false} />);
    expect(tree).toMatchSnapshot();
  });
});
