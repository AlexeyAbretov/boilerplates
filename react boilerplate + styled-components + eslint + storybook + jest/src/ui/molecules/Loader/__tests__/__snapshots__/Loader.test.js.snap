// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/atoms/Loader should be rendered 1`] = `
.c0 {
  position: relative;
}

.c0:before {
  content: '';
  display: block;
  padding-top: 100%;
}

.c0 .circular {
  -webkit-animation: GLFYz 2s linear infinite;
  animation: GLFYz 2s linear infinite;
  -webkit-transform-origin: center center;
  -ms-transform-origin: center center;
  transform-origin: center center;
  position: absolute;
  top: 0px;
  bottom: 0;
  right: 0px;
}

.c0 .path {
  stroke-dasharray: 1,200;
  stroke-dashoffset: 0;
  -webkit-animation: bgPtUW 1.5s ease-in-out infinite;
  animation: bgPtUW 1.5s ease-in-out infinite;
  -webkit-animation-play-state: running !important;
  animation-play-state: running !important;
  stroke-linecap: round;
  stroke: #52AE30;
}

<div
  className="c0"
  data-name="loader"
>
  <svg
    className="circular"
    viewBox="25 25 50 50"
  >
    <circle
      className="path"
      cx="50"
      cy="50"
      fill="none"
      r="20"
      strokeMiterlimit="10"
    />
  </svg>
</div>
`;

exports[`ui/atoms/Loader should be rendered as absolute 1`] = `
.c0 {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  width: 200px;
}

.c0:before {
  content: '';
  display: block;
  padding-top: 100%;
}

.c0 .circular {
  -webkit-animation: GLFYz 2s linear infinite;
  animation: GLFYz 2s linear infinite;
  -webkit-transform-origin: center center;
  -ms-transform-origin: center center;
  transform-origin: center center;
  position: absolute;
  top: 0px;
  bottom: 0;
  right: 0px;
}

.c0 .path {
  stroke-dasharray: 1,200;
  stroke-dashoffset: 0;
  -webkit-animation: bgPtUW 1.5s ease-in-out infinite;
  animation: bgPtUW 1.5s ease-in-out infinite;
  -webkit-animation-play-state: running !important;
  animation-play-state: running !important;
  stroke-linecap: round;
  stroke: #52AE30;
}

<div
  className="c0"
  data-name="loader"
  transform="translate(-50%, -50%)"
  width="200px"
>
  <svg
    className="circular"
    viewBox="25 25 50 50"
  >
    <circle
      className="path"
      cx="50"
      cy="50"
      fill="none"
      r="20"
      strokeMiterlimit="10"
    />
  </svg>
</div>
`;

exports[`ui/atoms/Loader should be rendered with overlay 1`] = `
Array [
  .c0 {
  background-color: rgb(153 153 153 / 60%);
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
  z-index: 9998;
}

<div
    className="c0"
    height="100%"
    width="100%"
  />,
  .c0 {
  position: relative;
  z-index: 9999;
}

.c0:before {
  content: '';
  display: block;
  padding-top: 100%;
}

.c0 .circular {
  -webkit-animation: GLFYz 2s linear infinite;
  animation: GLFYz 2s linear infinite;
  -webkit-transform-origin: center center;
  -ms-transform-origin: center center;
  transform-origin: center center;
  position: absolute;
  top: 0px;
  bottom: 0;
  right: 0px;
}

.c0 .path {
  stroke-dasharray: 1,200;
  stroke-dashoffset: 0;
  -webkit-animation: bgPtUW 1.5s ease-in-out infinite;
  animation: bgPtUW 1.5s ease-in-out infinite;
  -webkit-animation-play-state: running !important;
  animation-play-state: running !important;
  stroke-linecap: round;
  stroke: #52AE30;
}

<div
    className="c0"
    data-name="loader"
  >
    <svg
      className="circular"
      viewBox="25 25 50 50"
    >
      <circle
        className="path"
        cx="50"
        cy="50"
        fill="none"
        r="20"
        strokeMiterlimit="10"
      />
    </svg>
  </div>,
]
`;

exports[`ui/atoms/Loader should not be rendered 1`] = `null`;

exports[`ui/atoms/Loader should not be rendered without params 1`] = `null`;
